yubikey-u2f-pam-plugin/auth-pam-u2f.c

```diff
@@ -934,11 +934,12 @@ u2f_auth_verify(char *username, char* password, char* script_path, char **client
         buffer[br - 1] = '\0';
     }
     wait( &status);
+    close(pipefd[0]);
 
     if (WIFEXITED(status)) {
         if (client_reason != NULL && WEXITSTATUS(status) == 2) {
             fprintf(stderr, "AUTH-PAM: BACKGROUND: Got client reason %s\n", buffer);
-            *client_reason = &buffer;
+            memcpy(client_reason, buffer, sizeof(buffer));
```

Comment:
memcpy from our buffer into the callers passed array. This way we don't return a pointer to a stack allocated variable from this function, since upon returning this stack is technically no longer valid for use.